### ISSUE TITLE
Add check for new version availability

### ISSUE THREAD

**Issue by FutureAIGuru** [FutureAIGuru/BrainSimII#165]
**Is your feature request related to a problem? Please describe.**
When the program first starts, check to see if anew version is available and let the user know.

**Comment by Moorelife** [comment]
This would require that the program access some URL on GitHub or BrainSim.org to determine the most recent version. And of course this should be based on either a build date or a version number.

**Comment by FutureAIGuru** [comment]
It might be as simple as checking the date on the installer .exe file. Not sure.

**Comment by Moorelife** [comment]
You can do that, but that would still have to be the most recent installer file on the web, and you would not want to download it before you actually determine it is a newer version. So, we could do the following:
- Aside from the installer, also publish its MD5 checksum.
  (this is handy to determine whether the download is not tampered with as well).
- Have the installer calculate the same checksum for itself on installation
- On startup read the checksum stored by the installer as an indication of the version.
- If the read version and the installed version don't match, we can update.

**Comment by FutureAIGuru** [comment]
How about a webpage currentversion.html which contains the text of the current version. Program reads the html file, compares with current version, offers to update if html is greater.

It would be up to the release process to update currentversion.html.

**Comment by Moorelife** [comment]
That would be the simplest version, but leave the downloading and installing to the user....

**Comment by FutureAIGuru** [comment]
It turns out that the AssemblyVersion is not externally accessible but the AssemblyFileVersion IS. The problem is that AssemblyVersion will put on the build-time but the AssemblyFileVersion must be updated manually only. I have updated the buildAndSign batch file but not the buld_installer.cmd file to automatically post the assemblyFileVersion to the website. This means that the version must be manually bumped for every upload to the website  AND the help about should probably be modified to default to showing AssemblyFileVersion instead so it will match.

**Comment by Moorelife** [comment]
Then maybe using the MD5 hash is as good an approach, because that can automatically be computed

**Comment by FutureAIGuru** [comment]
I wrote a 3-line console app which can look at any .exe or .dll and get the version info if it exists and included the app in the installer build file. So when posting a new version to the download page, the version will also be written into a file "LatestBrainSimVersion.txt". This file can be read and compared to the running version. This is done in MainWindowExtension.cs and GetUpdateDialog.xaml. I included a "Don't check again" checkbox and property. 
This framework could be extended to create and check the hash.
